lib/cartesian_graph.dart

```diff
@@ -24,10 +24,11 @@ class CartesianGraph extends StatelessWidget{
   final List<Coordinates> Function(List<double>) coordinatesBuilder;
   final List<String> equations;
   GraphDisplay display;
+  final int chosenEquationIndex;
   final Color cursorColor;
 
 
-  CartesianGraph(this.bounds, {this.coordinates= const [], this.cursorLocation,this.legendColor = Colors.blueGrey, this.cursorColor = Colors.blue, this.lineColor = Colors.black, this.coordinatesBuilder, this.equations});
+  CartesianGraph(this.bounds, {this.coordinates= const [], this.cursorLocation, this.legendColor = Colors.blueGrey, this.cursorColor = Colors.blue, this.lineColor = Colors.black, this.coordinatesBuilder, this.equations, this.chosenEquationIndex = -1});
```

Comment:
this is a bit implementation specific, just make an Equation object that has a string for the equation and the color, then each line can be whatever color you want